bids/analysis/analysis.py

```diff
@@ -528,46 +495,38 @@ def setup_contrast(c):
 
         return self._contrasts
 
-    def matches_entities(self, entities, strict=False):
-        """Determine whether current AnalysisNode matches passed entities.
+    def get_model_spec(self, model=None):
+        """Return a ModelSpec instance for the current AnalysisNode.
 
         Parameters
         ----------
-        entities : dict
-            Dictionary of entities to match. Keys are entity
-            names; values are single values or lists.
-        strict : bool
-            If True, _all_ entities in the current Node must
-            match in order to return True.
-        """
-        return matches_entities(self, entities, strict)
-
+        model : dict
+            Optional dict containing BIDS-StatsModels model information. If
+            None provided, defaults to the model stored in the instance.
 
-def apply_transformations(collection, transformations, select=None):
-    """Apply all transformations to the variables in the collection.
+        Returns
+        -------
+        A bids.analysis.model_spec.ModelSpec instance.
 
-    Parameters
-    ----------
-    transformations : list
-        List of transformations to apply.
-    select : list
-        Optional list of names of variables to retain after all
-        transformations are applied.
-    """
-    for t in transformations:
-        kwargs = dict(t)
-        func = kwargs.pop('name')
-        cols = kwargs.pop('input', None)
-
-        if isinstance(func, str):
-            if func in ('and', 'or'):
-                func += '_'
-            if not hasattr(transform, func):
-                raise ValueError("No transformation '%s' found!" % func)
-            func = getattr(transform, func)
-            func(collection, cols, **kwargs)
-
-    if select is not None:
-        transform.Select(collection, select)
-
-    return collection
+        Notes
+        -----
+        If the current BIDSVariableCollection contains any sparse variables,
+        they will be automatically converted to dense before the ModelSpec
+        is constructed.
+        """
+        if model is None:
+            model = self.model
+        if self.model is None:
+            raise ValueError("Cannot generate a ModelSpec instance; no "
+                             "BIDS-StatsModels model specification found!")
+
+        if self.collection.all_dense():
+            collection = self.collection
+        else:
+            # TODO: move this logic to a to_dense() method in the Collection
+            collection = self.collection.clone()
+            sr = self.collection.sampling_rate
+            for name, var in collection.variables.items():
+                if isinstance(var, SparseRunVariable):
+                    collection.variables[name] = var.to_dense(sr)
+            return create_model_spec(collection, model)
```

Comment:
This led me down a short rabbit hole that made me realize none of this logic (I mean just in `get_model_spec` will actually work, because we can't assume that the model being constructed is for the run level (i.e., one might want to fit a model over subjects, in which case there's no concept of timing information).

I assume you have no objection if I add an optional `sampling_rate` argument to `get_model_spec()` that defaults to `TR`, but is only applied in the even that `collection.level == 'run'`? That should address both of our concerns.

Reply:
Yes, that sounds right.

Reply:
I've removed `get_design_matrix()`, but the above should now work if you do `collection.to_df(sampling_rate='TR')`.